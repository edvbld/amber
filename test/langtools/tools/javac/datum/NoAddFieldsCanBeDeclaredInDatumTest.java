/*
 * @test /nodynamiccopyright/
 * @summary smoke negative test for datum classes
 * @compile/fail/ref=NoAddFieldsCanBeDeclaredInDatumTest.out -XDrawDiagnostics NoAddFieldsCanBeDeclaredInDatumTest.java
 */

public class NoAddFieldsCanBeDeclaredInDatumTest {
    record Bad1(int i) {
        int y;
    }

    record Bad2(int i) {
        interface I {}
    }

    record Bad3(int i) {
        static {}
    }
}
