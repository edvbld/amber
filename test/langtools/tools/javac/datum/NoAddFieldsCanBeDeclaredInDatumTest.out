NoAddFieldsCanBeDeclaredInDatumTest.java:9:13: compiler.err.record.fields.must.be.in.header
NoAddFieldsCanBeDeclaredInDatumTest.java:9:14: compiler.err.expected: '('
NoAddFieldsCanBeDeclaredInDatumTest.java:13:9: compiler.err.illegal.start.of.type
NoAddFieldsCanBeDeclaredInDatumTest.java:13:19: compiler.err.invalid.meth.decl.ret.type.req
NoAddFieldsCanBeDeclaredInDatumTest.java:17:16: compiler.err.illegal.start.of.type
NoAddFieldsCanBeDeclaredInDatumTest.java:19:1: compiler.err.expected4: class, interface, enum, record
6 errors
